src/site/site.controller.ts

```diff
@@ -55,8 +63,18 @@ export class SiteController {
   @ApiOkResponse({ description: 'The site is successfully retrieved.', type: GetSiteDto })
   @ApiNotFoundResponse({ description: 'The site is not found.' })
   @UseInterceptors(new NotFoundInterceptor('The site is not found.'))
-  findOne(@Param('id', ParseIntPipe) id: number): ReturnType<SiteService['findOne']> | HttpException {
-    return this.siteService.findOne(id)
+  async findOne(@Param('id', ParseIntPipe) id: number): Promise<GetSiteDto | undefined> {
+    const site = await this.siteService.findOne(id)
+
+    if (!site) return undefined
+
+    const { DailySchedules, ...s } = site
+    const formattedSite = {
+      ...s,
+      schedule: this.scheduleService.toDto(DailySchedules)
+    }
+
+    return formattedSite
```

Comment:
Utilise `plainToClass` pour la safety de bien utiliser le dto 😉:

```suggestion
    return plainToClass(
      GetSiteDto,
      formattedSite
    )
```

Un example: https://github.com/compostage-dans-ma-ville/backend/blob/main/src/auth/auth.controller.ts#L46

Reply:
@Lyokolux pk il est marqué comme resolved alors que non? 🤔

Reply:
Un git push --force  a supprimé ce commit